dream_demo.ipynb

```diff
@@ -0,0 +1,261 @@
+    "# Display the detector position DataSet and calibration DataSet \n",
+    "print(\"Dataset with pixels initialized\")\n",
+    "print(Dream.data)\n",
+    "print(\"Calibration dataset:\")\n",
+    "print(Dream.cal)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generating a sample\n",
+    "A powder sample with a number of random Bragg peaks are generated. \n",
+    "\n",
+    "Here we create 30 Bragg peaks in the range of 2 to 10 AA with a $\\Delta$d over d range between 0.2 and 0.7 %. \n",
+    "\n",
+    "The relative strength of each of these is random as well, varying between 1.0 and 5.0. The sample data will be remembered by the Dream object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Dream.generate_random_sample(30, d_range=[2, 10], deltad_over_d_range=[0.2, 0.7], strength_range=[1.0, 5.0])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Generated bragg peak d spacings\") # The data can be accessed directly to check\n",
+    "print(Dream.sample_bragg_peaks_d)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generating random time of flight data\n",
+    "The fake instrument and sample are now used to generate a requested number of events. \n",
+    "\n",
+    "The wavelength window can be adjusted. \n",
+    "\n",
+    "A simple text based progress bar will be shown unless the keyword argument `verbose` is set to False. \n",
+    "\n",
+    "The resulting data is checked and compared to the ESS timeframe, it would be optimal to have just under 100% coverage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Generating random sparse data\")\n",
+    "Dream.generate_data_pseudo(n_events, wavelength_width=3.5, wavelength_center=3.0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(Dream.data) # Print an overview of the resulting dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert time of flight data to dspacing\n",
+    "The raw time of flight data now needs to be converted back to dspacing. This can be done using the calibration data calculated within the Dream object, or without assuming a perfect instrument. \n",
+    "\n",
+    "The two methods will give the same results in this case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dspacing = powder_reduction.powder_reduction(Dream.data, calibration=Dream.cal)\n",
+    "print(dspacing) # Show the converted dataset"
```

Comment:
Do you feel that comments like this are necessary in a notebook? Isn't it self documenting, given that the output will appear immediately below?

Reply:
It was reviewed internally without those comments, and more explanations were requested.